### PR TITLE
[mllama] Allow `pixel_values` with `inputs_embeds`

### LINKED ISSUE #38326: Allow `MllamaModel` to accept `pixel_values` and `inputs_embeds`
### Feature request

`MllamaModel` does not allow users to pass `pixel_values` and `inputs_embeds` simultaneously:
https://github.com/huggingface/transformers/blob/54cd86708d2b63a1f696ee1c59384a2f04100f57/src/transformers/models/mllama/modeling_mllama.py#L1702-L1705

However, commenting out those lines and running the follow script does generate the same logits:
```python
import torch
from transformers import MllamaForConditionalGeneration, AutoProcessor


model_id = "meta-llama/Llama-3.2-11B-Vision-Instruct"
model = MllamaForConditionalGeneration.from_pretrained(
    model_id, device_map="auto", torch_dtype=torch.bfloat16
)
processor = AutoProcessor.from_pretrained(model_id)

messages = [
    [
        {
            "role": "user",
            "content": [
                {
                    "type": "image",
                    "url": "https://llava-vl.github.io/static/images/view.jpg",
                },
                {"type": "text", "text": "What does the image show?"},
            ],
        }
    ],
]
inputs = processor.apply_chat_template(
    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
).to(model.device)

outputs = model(**inputs)

# Manually compute inputs_embeds
input_ids = inputs.pop("input_ids")
inputs_embeds = model.get_input_embeddings()(input_ids)
new_outputs = model(inputs_embeds=inputs_embeds, **inputs)
assert torch.allclose(outputs.logits, new_outputs.logits)
```

### Motivation

Being able to pass `inputs_embeds` along with `pixel_values` enables soft embeddings to be passed to the model in addition to images, which is useful for prompt tuning.

### Your contribution

Could contribute a PR removing the check assuming there isn't something I'm unaware of about the check.

### PR DESCRIPTION
**Pull request by dxoigmn**
# What does this PR do?

Allow the specification of both `pixel_values` and `inputs_embeds` at the same time in `MllamaModel.forward`.

Fixes #38326

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@zucchini-nlp

### CODE REVIEW

**Review by zucchini-nlp — COMMENTED** [review]
Thanks, left one tiny comment :)

**Review by zucchini-nlp — APPROVED** [review]
thanks
